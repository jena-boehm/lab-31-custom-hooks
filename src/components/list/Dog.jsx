import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Dog = ({ name, image }) => {
  return (
    <>
      <figure>
        <figcaption>{name}</figcaption>
        <Link to={`/breeds/${name}`}>
          <img src={image} alt={name} width="300px" />
        </Link>
      </figure>
    </>
  );
};

Dog.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string
};

export default Dog;
