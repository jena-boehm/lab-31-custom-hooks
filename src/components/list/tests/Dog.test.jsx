import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Dog from '../Dog';
import { MemoryRouter } from 'react-router-dom';

describe('Dog component', () => {
  afterEach(() => cleanup());
  it('renders Dog', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Dog />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
