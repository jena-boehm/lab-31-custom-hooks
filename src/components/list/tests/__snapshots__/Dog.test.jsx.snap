// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog component renders Dog 1`] = `
<DocumentFragment>
  <figure>
    <figcaption />
    <a
      href="/breeds/undefined"
    >
      <img
        width="300px"
      />
    </a>
  </figure>
</DocumentFragment>
`;
